src/platform/packages/shared/kbn-ebt-tools/src/performance_metrics/context/measure_interaction/index.ts

```diff
@@ -18,11 +18,13 @@ import { EventData } from '../performance_context';
 interface PerformanceMeta {
   queryRangeSecs: number;
   queryOffsetSecs: number;
+  isInitialLoad: boolean;
 }
 
 export function measureInteraction() {
   performance.mark(perfomanceMarkers.startPageChange);
-  const trackedRoutes: string[] = [];
+  const trackedRoutes = new Set<string>();
```

Comment:
Q: Do we remove the tracked routes at any point?

Aside from a potential memory leak, what happens if the user goes elsewhere in Kibana and comes back?

Would it be enough if `trackedRoutes (Set<string>)` is actually `currentPathname (string)`?

Reply:
The `trackedRoutes` updates with each route change. If the user navigates to another page (e.g., the services page) and then returns, `trackedRoutes` will be empty. 

The objective is to determine whether the current load is due to navigation (an initial load) or a page refresh. Happy to update it if you have a better approach. 

Not sure I understand the following, can you explain please

> Would it be enough if trackedRoutes (Set<string>) is actually currentPathname (string)?

Reply:
@afharo

Reply:
Tracking [`currentPathname`](https://github.com/elastic/kibana/blob/cd71ca903b6442165506a1ff0cb37e7b005c00a8/src/core/packages/chrome/browser-internal/src/project_navigation/project_navigation_service.ts#L269)  or plugin url would be better, see [comment](https://github.com/elastic/kibana/pull/206645#pullrequestreview-2559462032).